labman/gui/templates/quantification.html

```diff
@@ -28,40 +29,46 @@
   };
 
   $(document).ready(function(){
-    var concentrations = {% raw concentrations %};
+    var plates = {% raw plates %};
 
-    var sgOptions = {editable: true,
-                     enableAddRow: false  ,
+    var sgOptions = {editable: false,
+                     enableAddRow: false,
                      enableCellNavigation: true,
                      asyncEditorLoading: false,
                      enableColumnReorder: false,
                      autoEdit: false};
-    var sgCols = [{id: 'selector', name: '', field: 'header', width: 30}]
-    for (var i = 0; i < concentrations[0].length; i++) {
-      sgCols.push({id: i, name: i+1, field: i, editor: Slick.Editors.FloatEditor});
-    }
-    var data = [];
-    var rowId = 'A';
-    for (var i = 0; i < concentrations.length; i++) {
-      var d = (data[i] = {});
-      d["header"] = rowId;
-      for (var j = 0; j < concentrations[i].length; j++) {
-        d[j] = concentrations[i][j];
+    for (var plateInfo of plates) {
+      var concentrations = plateInfo['concentrations'];
+      var sgCols = [{id: 'selector', name: '', field: 'header', width: 30}]
+      for (var i = 0; i < concentrations[0].length; i++) {
+        sgCols.push({id: i, name: i+1, field: i});
+      }
+      var data = [];
+      var rowId = 'A';
+      for (var i = 0; i < concentrations.length; i++) {
+        var d = (data[i] = {});
+        d["header"] = rowId;
+        for (var j = 0; j < concentrations[i].length; j++) {
+          d[j] = concentrations[i][j];
```

Comment:
the python code uses 1-based indexing for row and column positions, but it looks as though this is using 0-based. is the client-side javascript code really using a different indexing scheme for well positions than server side? if yes, then that seems like a critical issue.

Reply:
Yes, they're and this is because of the use of a 3rd party library (slickgrid).